persistence/persistence-api/src/main/java/org/cbioportal/persistence/util/EhCacheUtils.java

```diff
@@ -0,0 +1,49 @@
+package org.cbioportal.persistence.util;
+
+
+import org.cbioportal.utils.config.annotation.ConditionalOnProperty;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.cache.CacheManager;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+@Component
+@ConditionalOnProperty(name = "persistence.cache_type", havingValue = {"ehcache-heap", "ehcache-disk", "ehcache-hybrid"})
+public class EhCacheUtils implements CacheUtils {
+    
+    @Autowired
+    private CustomEhcachingProvider customEhcachingProvider;
+    private CacheManager cacheManager;
+
+    @PostConstruct
+    public void init() {
+        this.cacheManager = customEhcachingProvider.getCacheManager();
+    }
+    
+    @Override
+    public List<String> getKeys(String cacheName) throws IllegalArgumentException {
+        javax.cache.Cache<String, Object> cache = cacheManager.getCache(cacheName);
+        if (cache == null) {
+            throw new IllegalArgumentException("Cannot find cache with name '" + cacheName + "'");
+        }
+        List<String> keysInCache = new ArrayList<String>();
+        Iterator<javax.cache.Cache.Entry<String, Object>> iterator = cache.iterator();
+        while (iterator.hasNext()) {
+            javax.cache.Cache.Entry<String, Object> entry = iterator.next();
```

Comment:
You can make this a for each if you want
```suggestion
        for (javax.cache.Cache.Entry<String, Object> entry : cache) {
            keysInCache.add(entry.getKey());
```

Reply:
Updated to:

```
        List<String> keysInCache = new ArrayList<>();
        cache.forEach(entry -> keysInCache.add(entry.getKey()));
        return keysInCache;
```